### ISSUE TITLE
Resource 'aws_iam_access_key.k' not found for variable 'aws_iam_access_key.k.encrypted_secret'

### ISSUE THREAD

**Issue by jamesgoodhouse** [terraform-providers/terraform-provider-aws#2350]
### Terraform Version
0.11.0

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_iam_access_key

### Terraform Configuration Files
```hcl
variable "create_user" { }
variable "public_key" { }

resource "aws_iam_user" "u" {
  count = "${var.create_user}"
  name  = "user_name"
  path  = "/"
}

resource "aws_iam_access_key" "k" {
  count   = "${var.create_user}"
  user    = "${aws_iam_user.u.name}"
  pgp_key = "${var.public_key}"
}

output "okta_user_encrypted_secret" {
  value = "${aws_iam_access_key.k.encrypted_secret}"
}
```

### Expected Behavior
`terraform plan` should not error, as it worked fine with 0.10.8.

### Actual Behavior
The error `output.okta_user_encrypted_secret: Resource 'aws_iam_access_key.k' not found for variable 'aws_iam_access_key.k.encrypted_secret'` happens.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. upgrade to terraform 0.11.0
2. `terraform plan`

**Comment by atsushi-ishibashi** [comment]
You use `count` param so maybe need to refer with index like `"${aws_iam_access_key.k.0.encrypted_secret}"` ?
```
resource "aws_iam_access_key" "k" {
  count   = "${var.create_user}"
```

**Comment by jamesgoodhouse** [comment]
Did something change in 0.11.0 with the count attribute, as all works fine with 0.10.8.

**Comment by atsushi-ishibashi** [comment]
@jamesgoodhouse When `1` is set to `var.create_user`, `plan` works well. In other cases, the below works well. Did you set the same value for `var.create_user` when you worked with 0.10.8?
```
variable "create_user" { }
variable "public_key" { }

resource "aws_iam_user" "u" {
  count = "${var.create_user}"
  name  = "user_name"
  path  = "/"
}

resource "aws_iam_access_key" "k" {
  count   = "${var.create_user}"
  user    = "${element(aws_iam_user.u.*.name, count.index)}"
  pgp_key = "${var.public_key}"
}

output "okta_user_encrypted_secret" {
  value = "${aws_iam_access_key.k.*.encrypted_secret}"
}
```

**Comment by kwach** [comment]
This might be related: https://github.com/hashicorp/terraform/issues/16688
Should this not be tracked here I'll create separate issue.

**Comment by jamesgoodhouse** [comment]
@atsushi-ishibashi: yep, i'm using the same value as I did for 0.10.8. Literally nothing has changed on my end other than upgrading to 0.11.0.

**Comment by jamesgoodhouse** [comment]
I was pointed to this as the culprit of my issues. I've yet to make changes to test it out, but looks promising.

https://www.terraform.io/upgrade-guides/0-11.html#error-checking-for-output-values

**Comment by radeksimko** [comment]
Just confirmed (by trying to `apply` the attached config with latest build from current `master`) this is indeed a regression which was fixed in #2348 and will be part of the next release.

Sorry for any inconvenience.
